src/logcollector/logcollector.c

```diff
@@ -707,14 +701,25 @@ int handle_file(int i, int j, int do_fseek, int do_log)
             merror(FSEEK_ERROR, lf->file, errno, strerror(errno));
             fclose(lf->fp);
             lf->fp = NULL;
-            return (-1);
+            goto error;
         }
 #endif
     }
 
     /* Set ignore to zero */
     lf->ign = 0;
     return (0);
+
+error:
+    lf->ign++;
+
+    if (open_file_attempts && j < 0) {
+        mdebug1(OPEN_ATTEMPT, lf->file, open_file_attempts - lf->ign);
```

Comment:
When trying to open a monitored file which contains a `strftime` format. The count fails by reaching negative values:

```
2018/12/12 09:44:23 ossec-logcollector[121168] logcollector.c:385 at LogCollectorStart(): DEBUG: (1962): Unable to open file '/root/test-timestamp/file.log-2018-12-12'. Remaining attempts: 0
2018/12/12 09:44:23 ossec-logcollector[121168] logcollector.c:509 at LogCollectorStart(): INFO: (1904): File not available, ignoring it: '/root/test-timestamp/file.log-2018-12-12'.
2018/12/12 09:44:28 ossec-logcollector[121168] logcollector.c:717 at handle_file(): DEBUG: (1962): Unable to open file '/root/test-timestamp/file.log-2018-12-12'. Remaining attempts: -997
2018/12/12 09:44:33 ossec-logcollector[121168] logcollector.c:717 at handle_file(): DEBUG: (1962): Unable to open file '/root/test-timestamp/file.log-2018-12-12'. Remaining attempts: -998
```

Reply:
Fixed. I made Logcollector check the attempt limit when handling files with timestamp pattern.